### PR TITLE
Rewrote the promise shim to be usable in a node environment.

### PR DESCRIPTION
**Pull request by yomexzo**
👀 @tessalt @tanema @minasmart

### CODE REVIEW

File: shims/promise.js [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
+
+let RSVP;
+
+if (isNodeLikeEnvironment()) {
+  const localRequire = require;
+
+  RSVP = localRequire('rsvp');
+} else {
+  RSVP = window.RSVP;
+}
```
Comment (minasmart) [inline shims/promise.js]: So, weird thing about this: I'm pretty sure nothing is actually using this shim? Nothing imports promise or RSVP. What exactly needs it?

Comment (minasmart) [inline shims/promise.js]: Can we just remove the shim altogether?

Comment (yomexzo) [inline shims/promise.js]: hmm... They are used in tests [here](https://github.com/Shopify/js-buy-sdk/blob/master/tests/unit/adapters/listings-adapter-test.js#L4) and [here](https://github.com/Shopify/js-buy-sdk/blob/master/tests/unit/api/shop-client-test.js#L5)

Comment (minasmart) [inline shims/promise.js]: Shouldn't be. Just remove the import and use the global promise. Then this shim can go away 😄

Comment (yomexzo) [inline shims/promise.js]: I'm thinking, Won't that break in environments like IE10?

Comment (minasmart) [inline shims/promise.js]: no, because it's polyfilled still. This is just a shim so that you can `import`

Comment (yomexzo) [inline shims/promise.js]: i see it. thanks!

**Comment by yomexzo** [comment]
Will be creating another pull request to delete the promise shim

**Pull request closed without merging**